imagenet/main.py

```diff
@@ -48,18 +49,66 @@
                     help='evaluate model on validation set')
 parser.add_argument('--pretrained', dest='pretrained', action='store_true',
                     help='use pre-trained model')
+parser.add_argument('--finetune', dest='finetune', action='store_true',
+                    help='fine tune pre-trained model')
 
 best_prec1 = 0
 
+class FineTuneModel(nn.Module):
+    def __init__(self, original_model, arch, num_classes):
+        super(FineTuneModel, self).__init__()
+
+        if arch.startswith('alexnet') :
+            self.features = original_model.features
+            self.classifier = nn.Sequential(
+                nn.Dropout(),
+                nn.Linear(256 * 6 * 6, 4096),
+                nn.ReLU(inplace=True),
+                nn.Dropout(),
+                nn.Linear(4096, 4096),
+                nn.ReLU(inplace=True),
+                nn.Linear(4096, num_classes),
+            )
+            self.modelName = 'alexnet'
+        elif arch.startswith('resnet') :
+            # Everything except the last linear layer
+            self.features = nn.Sequential(*list(original_model.children())[:-1])
+            self.classifier = nn.Sequential(
+                nn.Linear(512, num_classes)
```

Comment:
`nn.Linear(512, num_classes)` should be replaced by
`original_model.fc.in_features` to be generic to all resnets

Reply:
This is a good example to extract the weight parameters of the specific layer.